misk/src/main/kotlin/misk/web/actions/NotFoundAction.kt

```diff
@@ -20,44 +21,35 @@ import javax.inject.Provider
 import javax.inject.Singleton
 
 @Singleton
-class NotFoundAction @Inject constructor(
-  @JvmSuppressWildcards private val clientHttpCall: ActionScoped<HttpCall>
+class NotFoundAction @Inject internal constructor(
+  @JvmSuppressWildcards private val clientHttpCall: ActionScoped<HttpCall>,
+  private val servletProvider: Provider<WebActionsServlet>
 ) : WebAction {
-  @Inject internal lateinit var servletProvider: Provider<WebActionsServlet>
-
   @Get("/{path:.*}")
   @Post("/{path:.*}")
   @RequestContentType(MediaTypes.ALL)
   @ResponseContentType(MediaTypes.ALL)
   @Unauthenticated
   fun notFound(@PathParam path: String): Response<ResponseBody> {
+    val httpCall = clientHttpCall.get()
+    val alternativeActions = urlOnlyMatches(httpCall)
+    // If there are any actions that match on URL the client is likely using the wrong
+    // content type, so we'll return a "415 Unsupported Media Type".
+    val statusCode = if (alternativeActions.isNotEmpty()) 415 else 404
```

Comment:
Proposal: what if we also checked the HTTP verb (`httpCall.dispatchMechanism`) to see if we have a bound action for another verb and, if so, return `405 Method Not Allowed`?